homeassistant/components/ads/binary_sensor.py

```diff
@@ -34,14 +20,20 @@ def setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the Binary Sensor platform for ADS."""
+    entities = []
+
+    if discovery_info is None:
```

Comment:
Please move this check up at the top of the function body. We don't need to do anything until this check is true.